### PR TITLE
libscf_solver LGTM

### PR DESCRIPTION
**Pull request by JonathonMisiewicz**
The libscf_solver segment of #1615 in what should be its entirety. Mostly changing format strings, but also setting a header guard back to guarding the header, rather than a blank line. Passes quicktests.

### CODE REVIEW

File: psi4/src/psi4/libscf_solver/hf.cc [hunk 1]
```diff
@@ -769,7 +769,7 @@ void HF::form_Shalf() {
             // descending order
             int start_index = 0;
             for (int i = 0; i < dimpi[h]; ++i) {
-                if (S_cutoff < eigval->get(h, i)) {
+                if (S_cutoff < eigval->get(h, i)) { // Include this empty block by analogy of prev. block
```
Comment (loriab) [inline psi4/src/psi4/libscf_solver/hf.cc]: what do others think? (a) comment (like above change), (b) `continue`, (c) suppress lgtm, (d) `<` --> `>=`. I'm thinking (d)

Comment (andysim) [reply psi4/src/psi4/libscf_solver/hf.cc]: Yup, d) gets my vote

Comment (andysim) [reply psi4/src/psi4/libscf_solver/hf.cc]:
```suggestion
                if (S_cutoff >= eigval->get(h, i)) {
```

File: psi4/src/psi4/libscf_solver/hf.cc [hunk 2]
```diff
@@ -769,7 +769,7 @@ void HF::form_Shalf() {
...
                 } else {
```
Comment (andysim) [inline psi4/src/psi4/libscf_solver/hf.cc]:
```suggestion
```

**Pull request merged**